inferior-idris.el

```diff
@@ -368,4 +368,30 @@ call `ERROR' if there was an Idris error."
       ((:error condition &optional _spans)
        (message "Setting option %s to %s returned an error: %s." opt b condition)))))
 
+(defun idris-get-idris-version ()
+  "Ask the Idris compiler for its version information.
+Returns a cons cell whose car is a list of version number
+components and whose cdr is a list of prerelease identifiers, if
+applicable. Returns nil if the version of Idris used doesn't
+support asking for versions."
+  (pcase (idris-eval :version t)
+    (`((,version ,prerelease)) (cons version prerelease))
+    (t nil)))
+
+(defun idris-get-idris-version-string ()
+  "Ask the Idris compiler for its version information, and return the result as a user-friendly string.
+Returns nil if the version of Idris used doesn't support asking
+for versions."
+  (accept-process-output idris-process 0.1) ; work around race condition
```

Comment:
It's a race condition in Emacs, so far as I could tell - I couldn't trigger it reliably, but it didn't happen after I added this line. I'd get back that Idris replied to an invalid message ID. Let me do some more poking.